src/goose/toolkit/developer.py

```diff
@@ -91,6 +94,46 @@ def update_plan(self, tasks: List[dict]) -> List[dict]:
         # Return the tasks unchanged as the function's primary purpose is to update and display the task status.
         return tasks
 
+    @tool
+    def fetch_web_content(self, url: str) -> str:
+        """
+        Fetch content from a URL using available tools. Attempts to use playwright first, then falls back to wget, curl, or httpx.
+
+        Args:
+            url (str): a name of a file with the url/page content. This may be large so you can use search tools to find content. It may also contain other links which can be browsed.
+        """  # noqa
+        try:
+            with sync_playwright() as p:
+                browser = p.chromium.launch()
+                page = browser.new_page()
+                page.goto(url)
+                content = page.content()
+                browser.close()
+                with tempfile.NamedTemporaryFile(delete=False, mode="w", suffix=".txt") as tmp_file:
+                    tmp_file.write(content)
+                    return tmp_file.name
+        except Exception:
+            self.notifier.log(
```

Comment:
Might be something wrong with my setup, but I wasn't getting this log message. After asking again, goose told me chromium does not exist.

Reply:
hrm ok - will take a look

Reply:
@zakiali did it fall back to curl?

Reply:
yes was a typo, wasn't logging correctly and this goose capturing the failure.